analysers/analyser_osmosis_highway_deadend.py

```diff
@@ -366,6 +369,7 @@
     JOIN nodes ON
       nodes.id = results_recursive.nid
 ) AS t
+JOIN oneway_highway ON wid = oneway_highway.id
```

Comment:
> One thing to be checked on a larger extract: the id of oneway_highway isn't indexed, so it may be faster to use table highways instead to JOIN, which already has an index (without needing to build another index, which also costs time).

Current:
```
['Unique  (cost=9046767.12..9078631.33 rows=200 width=80) (actual time=431.099..431.337 rows=203 loops=1)']
['  ->  Sort  (cost=9046767.12..9062699.23 rows=6372842 width=80) (actual time=431.098..431.180 rows=1634 loops=1)']
['        Sort Key: results_straightforward.nid, results_straightforward.wid']
['        Sort Method: quicksort  Memory: 278kB']
['        ->  Merge Join  (cost=82826.05..8144485.78 rows=6372842 width=80) (actual time=370.268..430.595 rows=1634 loops=1)']
['              Merge Cond: (results_straightforward.wid = oneway_highway.id)']
['              ->  Sort  (cost=6857.00..6864.30 rows=2920 width=48) (actual time=1.434..1.455 rows=203 loops=1)']
['                    Sort Key: results_straightforward.wid']
['                    Sort Method: quicksort  Memory: 40kB']
['                    ->  Append  (cost=0.00..6688.93 rows=2920 width=48) (actual time=0.027..1.398 rows=203 loops=1)']
['                          ->  Seq Scan on results_straightforward  (cost=0.00..20.70 rows=1070 width=48) (actual time=0.027..0.048 rows=198 loops=1)']
['                          ->  Nested Loop  (cost=0.56..6635.13 rows=1850 width=48) (actual time=0.017..1.332 rows=5 loops=1)']
['                                ->  Seq Scan on results_recursive  (cost=0.00..28.50 rows=1850 width=16) (actual time=0.001..0.003 rows=5 loops=1)']
['                                ->  Index Scan using pk_nodes on nodes  (cost=0.56..3.57 rows=1 width=40) (actual time=0.264..0.264 rows=1 loops=5)']
['                                      Index Cond: (id = results_recursive.nid)']
['              ->  Sort  (cost=75969.05..77060.29 rows=436496 width=40) (actual time=368.341..389.638 rows=436151 loops=1)']
['                    Sort Key: oneway_highway.id']
['                    Sort Method: quicksort  Memory: 41850kB']
['                    ->  Seq Scan on oneway_highway  (cost=0.00..35078.96 rows=436496 width=40) (actual time=0.024..248.841 rows=436496 loops=1)']
['Planning Time: 0.252 ms']
['Execution Time: 435.084 ms']
```

Using `highways` instead of `oneway_highway` (here and on line 354):
```
['Unique  (cost=19632.60..19647.20 rows=200 width=58) (actual time=78.626..78.679 rows=203 loops=1)']
['  ->  Sort  (cost=19632.60..19639.90 rows=2920 width=58) (actual time=78.625..78.638 rows=203 loops=1)']
['        Sort Key: results_straightforward.nid, results_straightforward.wid']
['        Sort Method: quicksort  Memory: 53kB']
['        ->  Nested Loop  (cost=0.43..19464.53 rows=2920 width=58) (actual time=0.868..78.426 rows=203 loops=1)']
['              ->  Append  (cost=0.00..6688.93 rows=2920 width=48) (actual time=0.028..0.505 rows=203 loops=1)']
['                    ->  Seq Scan on results_straightforward  (cost=0.00..20.70 rows=1070 width=48) (actual time=0.027..0.108 rows=198 loops=1)']
['                    ->  Nested Loop  (cost=0.56..6635.13 rows=1850 width=48) (actual time=0.024..0.366 rows=5 loops=1)']
['                          ->  Seq Scan on results_recursive  (cost=0.00..28.50 rows=1850 width=16) (actual time=0.009..0.011 rows=5 loops=1)']
['                          ->  Index Scan using pk_nodes on nodes  (cost=0.56..3.57 rows=1 width=40) (actual time=0.069..0.069 rows=1 loops=5)']
['                                Index Cond: (id = results_recursive.nid)']
['              ->  Index Scan using idx_highways_id on highways  (cost=0.43..3.12 rows=1 width=18) (actual time=0.380..0.380 rows=1 loops=203)']
['                    Index Cond: (id = results_straightforward.wid)']
['Planning Time: 0.272 ms']
['Execution Time: 78.761 ms']
```

Seems to be more efficient to use table `highways`, using the existing index

Reply:
Like this https://github.com/osm-fr/osmose-backend/pull/2176/commits/813e0c9110ee1926173949151aca758f7001b687 ?

Reply:
Just use `highways.highway` instead of `highways.tags->'highway'` :)
See my suggestion